### ISSUE TITLE
Not all search options always show up in the pop-up menu

### ISSUE THREAD

**Issue by Bortseb** [Pitmairen/selection-search#1]
Is there a reason why? it seems to have only started doing this recently for me.

**Comment by Pitmairen** [comment]
I haven't noticed this. Are the searches permanently not showing or is it just sometimes?  And is it always the same search engines that are not showing?

There is an option tough on each search that can be checked to make the search show only in the popup or contextmenu if the global "separate search engines" option is enabled.

**Comment by Bortseb** [comment]
It only happens sometimes, and it seems to have just started happening to me recently. For example, I tried to open a selection search on some text right here in Github, but only 2 of my 3 options were shown.

The one that is always missing is this one:
https://plus.google.com/u/0/s/%s
For a google+ search.

I have all of my 3 searches opening in both the pop-up and the context menu, and i don't have the other option you are referring to checked off. I use middle mouse click to activate the popup... but I tired different options just now and the same bugs is still happening. I use default styling for the popup.

Some other weird stuff I've experienced is that none of the favicons for my searches are right. They are just generic globes... Plus sometime they force the search engine text down onto a second line making the popup twice as long. I wish there was a sytling option for the popup that just showed the search engine name... no favicon.

Anyhow... I still love the extension! It quickly became part of how I surf the web.

**Comment by Pitmairen** [comment]
If you put the following css into the styling box in popup menu -> advanced settings -> customize, the icons should be hidden:
```
#popup img{
 display: none;
}
```

Do you get any error messages in the console (ctrl+shift+j) when the the search is not showing?
Or have you installed any new extensions recently?

**Comment by Bortseb** [comment]
Thanks for the CSS tip! That works better! 

Also... It seems that my G+ search disappeared because of an ad blocker I recently installed.
So thanks for that too!

**Comment by Pitmairen** [comment]
In case you didn't see the edit of the previous post there is an option "Always remove icons" in the advanced settings which completely removes the icons and not only hides them.

So it's the adblock that is causing the problem?

**Comment by Bortseb** [comment]
Yes it was. I disabled it, and then Google+ popped back up in the list.

**Comment by Pitmairen** [comment]
The url can probably be white listed in the settings of the adblocker or something. 
I haven't had any such problems with adblock plus.

**Comment by Bortseb** [comment]
One final Question Since I have your attention... (Sorry, this is way off the original issue topic)

I use this other extension to select the text of hyperlinks:
https://chrome.google.com/webstore/detail/hyperlink-text-selector/ojnfanikhkhoklphdcehbolcpeipbaec

Obviously I use this in combination with your extension often.... However, there is one case where it doesn't quite work...

Sometimes when I select hyperlink text, and then use middle click to activate Selection Search, the copy to clipboard option doesn't work. The cases when this happens is when, by middle-clicking, the selection of the hyperlink text is lost. (obviously this is an issue with the other extension, not yours, but you will see why I think, maybe, my problem can be fixed on your end) By "lost" i mean it is no longer highlighted in blue. (when I do this with regular text, the selection isn't lost by middle clicking) As long as the selection isn't lost, the clipboard option seems to work fine.

What is really strange, even though the selection is "lost" Selection Search still works perfectly fine on all my search engine options... It is only the clipboard option that doesn't work as expected.

Here is how to duplicate....
1. I go to google and do a search.
2. I use Hyperlink Text Selector to select the text of any of the many links in the results.
3. I click all 3 of my Selection Search options in the popup
4. All searches work fine, but the clipboard option puts this in my clipboard:
"https://www.google.ca/COPY"... This is not the Selected text of course...

Basically, its just takes the root domain of whatever page I am on, and then adds COPY to the end...

Other than this special case, the clipboard option works fine for me.

Any idea why this might be happening?

**Comment by Bortseb** [comment]
To add to the last question... The domain that is added to the clipboard is the one of the page I am on, not of the link that I selected the text of.

**Comment by Bortseb** [comment]
And, adblock is blocking this option because I had "Fanboy's Social Blocking List" turned on.

**Comment by Pitmairen** [comment]
First of all thaks for letting me know about that extension, i have always wanted to select text in link but i didn't know it was possible. That extension works great.

But i can't seem to reproduce your problem. When clicking the link the selection is not lost. 

The only difference between the searche engines and the copy option is that the selection is read and added to the search engine when the mouse enters the link, but with the copy option the selection is not read until the link is clicked. 

It could be that it behaves differently on different operating systems. I'm currently running linux. What os are you using?

**Comment by Bortseb** [comment]
I fixed it... it was only because I was using middle click... I switched to Ctrl+Shift+Left click and it works fine now.

I guess its just because of how the browser deals with a middle click differently.

**Comment by Bortseb** [comment]
One more question... Would it be possible to have multiple triggers for the pop up, instead of just one? I much prefer to use middle click... but now that I know its flawed... I want another option to be ther at all times.

**Comment by Pitmairen** [comment]
It should be possible, but the code needs to be modifies. I will add it when i have time.

**Comment by Pitmairen** [comment]
I have added an option to allow for multiple activators. You can test it by loading the extension as a unpacked extension on the extension page in developer mode.

**Comment by Bortseb** [comment]
Can I get more instructions.... I tried... It asked me to find it on my computer... I am being told it can't find manifest file...

Do I need to download something?

**Comment by Pitmairen** [comment]
You download this: https://github.com/Pitmairen/selection-search/archive/master.zip  and extract it somewhere.

Then go to chrome://extensions/ and check the "Developer mode" checkbox at the top. Then click "Load unpacked extension" and select the folder you extracted. That should be it, the extension should be loaded and ready for use. You will now have two versions of the extension so you should probably disable the version downloaded from webstore while testing so it doesn't interfere. 

If it's working i will upload  a new version to the web store.

**Comment by Bortseb** [comment]
Ok, tested. It works for me!

**Comment by Pitmairen** [comment]
I forgot to mention that you also have to compile the templates to make the options page work correctly, but then you will need to install some stuff. I will include the compiled templates in the repository later.

**Comment by Pitmairen** [comment]
I have uploaded a new version now.

**Comment by Droyk** [comment]
[Pitmairen](https://github.com/Pitmairen) close this issue it now seem to be fixed
